### PR TITLE
Fixed undefined method error when doing http basic authentication.

### PR DESCRIPTION
**Pull request by victorteokw**
Fixed error `undefined method `split' for nil:NilClass` when

``` ruby
name, password = ActionController::HttpAuthentication::Basic.user_name_and_password(request)
```

when `request.authorization` returns nil, `auth_param` and `auth_scheme` raise error.

### CODE REVIEW

File: actionpack/lib/action_controller/metal/http_authentication.rb [hunk 1]
```diff
@@ -106,11 +106,11 @@ def decode_credentials(request)
       end
 
       def auth_scheme(request)
-        request.authorization.split(' ', 2).first
+        (request.authorization || '').split(' ', 2).first
```
Comment (egilburg) [inline actionpack/lib/action_controller/metal/http_authentication.rb]: Can do `request.authorization.to_s.split('  ', 2).first`

`.to_s` will not make a new object if orginal is already a string, so shouldn't be a performance issue.

Comment (cheunghy) [inline actionpack/lib/action_controller/metal/http_authentication.rb]: Thanks.

Will it be a performance issue to evaluate ``request.authorization`` twice?
``` ruby
request.authorization.split(' ', 2).first if request.authorization
```

**Comment by senny** [comment]
@cheunghy could you add a test-case for this?

**Comment by cheunghy** [comment]
@senny Test added.

**Comment by rafaelfranca** [comment]
Could you open the pull request against master branch? We may decide to not backport to 4-2-stable.

**Comment by cheunghy** [comment]
https://github.com/rails/rails/pull/18666

**Comment by sgrif** [comment]
There is no need to open pull requests against 4-2-stable, we will backport the merge commit from a PR made against master. This also includes the entire master branch.

**Comment by cheunghy** [comment]
Thanks!
Yes, I have already move it to https://github.com/rails/rails/pull/18666.

**Pull request closed without merging**